### PR TITLE
Port splitCheck and SplitCheckObserver implementations, step 2

### PR DESCRIPTION
**Pull request by xiaobogaga**
@coocood PTAL, thanks.

### CODE REVIEW

File: tikv/raftstore/coprocessor.go [hunk 1]
```diff
@@ -5,6 +5,7 @@ import (
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/pingcap/kvproto/pkg/raft_cmdpb"
+	"github.com/prometheus/common/log"
```
Comment (coocood) [inline tikv/raftstore/coprocessor.go]: imported the wrong package.

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,8 @@
 module github.com/ngaut/unistore
 
 require (
+	github.com/alecthomas/template v0.0.0-20160405071501-a0175ee3bccc // indirect
```
Comment (coocood) [inline go.mod]: introduced by wrong log package "github.com/prometheus/common/log"

File: tikv/raftstore/worker.go [hunk 3]
```diff
@@ -270,6 +273,8 @@ func (r *splitCheckRunner) scanSplitKeys(spCheckerHost *splitCheckerHost, region
 			return nil, errors.Trace(err)
 		}
 	}
+	// Close the reader early, and in the `onKv` method above, some keys must be copied.
+	reader.Close()
```
Comment (coocood) [inline tikv/raftstore/worker.go]: The reader doesn't hold any lock, we don't need to close it early.

File: tikv/raftstore/coprocessor.go [hunk 4]
```diff
@@ -68,21 +69,56 @@ func newSizeSplitChecker(maxSize, splitSize, batchSplitLimit uint64, policy pdpb
...
 
 func (checker *sizeSplitChecker) getSplitKeys() [][]byte {
-	//Todo, currently it is a place holder
+	// Make sure not to split when less than maxSize for last part
+	if checker.currentSize+checker.splitSize < checker.maxSize {
+		if len(checker.splitKeys) != 0 {
+			checker.splitKeys = checker.splitKeys[:len(checker.splitKeys)-1]
+		}
+	}
+	if len(checker.splitKeys) != 0 {
```
Comment (coocood) [inline tikv/raftstore/coprocessor.go]: We can safely remove this check.

File: tikv/raftstore/coprocessor.go (hunk 1 above)
Comment (xiaobogaga) [reply tikv/raftstore/coprocessor.go]: ok, thanks

File: tikv/raftstore/worker.go (hunk 3 above)
Comment (xiaobogaga) [reply tikv/raftstore/worker.go]: yes, thanks.

File: tikv/raftstore/coprocessor.go (hunk 4 above)
Comment (xiaobogaga) [reply tikv/raftstore/coprocessor.go]: yes, thanks.

**Comment by xiaobogaga** [comment]
@coocood PTAL again, thanks.

**Comment by coocood** [comment]
LGTM

**Pull request merged**